test/integration/component/test_ospf.py

```diff
@@ -0,0 +1,477 @@
+        self.batman_vm_1 = VirtualMachine.create(
+            self.apiclient,
+            self.services["batman_vm_1"],
+            serviceofferingid=self.service_offering.id,
+            ipaddress="200.100.1.101",
+            networkids=self.batman_network_tier1.id
+        )
+        self.logger.debug("Deployed VM %s in network: %s" % (self.batman_vm_1.id, self.batman_network_tier1.id))
+        
+        self.services["superman_vm_1"]["zoneid"] = self.zone.id
+        self.services["superman_vm_1"]["template"] = self.template.id
+        self.superman_vm_1 = VirtualMachine.create(
+            self.apiclient,
+            self.services["superman_vm_1"],
+            serviceofferingid=self.service_offering.id,
+            ipaddress="200.100.5.101",
+            networkids=self.superman_network_tier1.id
+        )
+        self.logger.debug("Deployed VM %s in network: %s" % (self.superman_vm_1.id, self.superman_network_tier1.id))
+        
+        """ check ping from VR of batman to vm of superman""" 
+        self.batman_vm_pip = PublicIPAddress.create(
+            self.apiclient,
+            zoneid=self.zone.id,
+            networkid=self.batman_network_tier1.id,
+            vpcid=self.batman_vpc.id
+        )
+        self.logger.debug("Acquired public ip for batman vm %s" % self.batman_vm_pip.ipaddress.ipaddress )
+    
+        StaticNATRule.enable(
+            self.apiclient,
+            ipaddressid= self.batman_vm_pip.ipaddress.id,
+            virtualmachineid=self.batman_vm_1.id,
+            networkid=self.batman_network_tier1.id
+        )
+        
+        self.logger.debug("Static Nat enabled on ip %s" % self.batman_vm_pip.ipaddress.ipaddress )
+        
+        ssh = SshClient(self.batman_vm_pip.ipaddress.ipaddress, 22, "root", "password") 
+        res = ssh.execute("ping -c 1 200.100.5.101")
+        print res
+        result = str(res)
+        self.assertEqual(
+            result.count("0% packet loss"),
+            1,
+            "Ping to other VM @ 200.100.5.101 succeeded"
+        )
+        res = ssh.execute("ping -c 1 200.100.1.101")
+        print res
+        result = str(res)
+        self.assertEqual(
+            result.count("0% packet loss"),
+            1,
+            "Ping to other VM @ 200.100.5.101 succeeded"
+        )
+        
+        self.logger.debug("Test successful clearing VMs")
+        self.cleanup.append(self.batman_vm_1)
+        self.cleanup.append(self.superman_vm_1)
+        self.cleanup.append(self.service_offering)
+        self.logger.debug("Test successful clearing VPC Tiers")
+        self.cleanup.append(self.superman_network_tier1)
+        self.cleanup.append(self.batman_network_tier1)
+        self.cleanup.append(self.network_offering)
+        self.logger.debug("Test successful clearing VPCs")
+        self.cleanup.append(self.superman_vpc)
+        self.cleanup.append(self.batman_vpc)
+        self.cleanup.append(self.vpc_off)
```

Comment:
the resources added to the cleanup list are picked up by Marvin when tearDown is called.